File: GFG_Scala/src/org/main/gfg/scala/algo/mathematics/PalindromeNumber.scala
package org.main.gfg.scala.algo.mathematics

import java.util._
import scala.annotation.tailrec

object PalindromeNumber extends App {
  val scan = new Scanner(System.in)
  val number = scan.nextInt
  var sum = 0
  var temp = number
  while(temp != 0) {
    sum = sum * 10 + temp%10
    temp = temp/10
  }
  println(sum == number)
  
  @tailrec
  def palindromeNo(n: Int, sum: Int = 0): Int = {
	  if (n / 10 == 0) sum * 10 + n
	  else palindromeNo(n / 10, sum * 10 + n % 10)
  }
  
  println(palindromeNo(number))
  println(palindromeNo(number) == number)
}

File: GFG_Java/src/org/main/gfg/java/algo/mathematics/PalindromeNumber.java
package org.main.gfg.java.algo.mathematics;

import java.util.*;

public class PalindromeNumber {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int number = scan.nextInt();
		int sum = 0, number1 = number;
		while(number1 != 0) {
			sum = sum * 10 + number1%10;
			number1 = number1/10;
		}
		System.out.println(sum == number);
		System.out.println(palindrome(number, 0) == number);
		scan.close();
		
	}
	
	static int palindrome(int number, int sum) {
		if (number / 10 == 0) return (sum * 10 + number);
		return palindrome(number / 10, sum * 10 + number % 10);
	}
}
